### PR TITLE
Python: update metadata.

### PR DESCRIPTION
**Pull request by landrito**
Updates #1320:

Does the following

- Remove `requirements.txt` entirely.
- Remove `oauth2client` from the dependency list in `setup.py`; add `google-auth` and `requests`.
- Remove dependencies on `proto-google-cloud-*`.
- Add `google.cloud.proto`, `google.cloud.proto.{api}` to `namespace_packages`.
- Add Python 3.6 to the list of tested Python versions.

Need to be done:

- [x] Update the auth version in googleapis.
 - https://github.com/googleapis/googleapis/pull/435

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java [hunk 1]
```diff
@@ -198,6 +203,50 @@ private ViewModel generateMetadataView(
...
+            "google-gax", packageConfig.gaxVersionBound(TargetLanguage.PYTHON)));
+    dependencies.add(
+        PackageDependencyView.create(
+            "google-auth", packageConfig.authVersionBound(TargetLanguage.PYTHON)));
+    dependencies.add(
+        PackageDependencyView.create("requests", VersionBound.create("2.18.4", "3.0dev")));
+    return dependencies.build();
+  }
+
+  private boolean hasLongrunningMethods(Model model, GapicProductConfig productConfig) {
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java]: This appears to be unused.

File: src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java [hunk 2]
```diff
@@ -267,6 +316,8 @@ private OptionalArrayMethodView createExampleApiMethodView(GapicMethodContext co
         namespacePackages.add(subPackage);
       }
     }
+    namespacePackages.add("google.cloud.proto");
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java]: This adds `google.cloud.proto` but not `google.cloud.gapic`.

Comment (landrito) [inline src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java]: Google.cloud.gapic was switched over to the new path "google.cloud.api_vx.gapic" and that should be computed already.

Comment (landrito) [inline src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java]: Updated the top comment. Sorry forgot to remove those after copy pasting.

File: src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java (hunk 1 above)
Comment (landrito) [inline src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java]: Oops thats for a different PR. Fixed.

**Pull request merged**